radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java

```diff
@@ -67,77 +67,111 @@
 import static com.google.common.primitives.UnsignedBytes.lexicographicalComparator;
 
 import com.google.common.base.Stopwatch;
+import com.google.common.base.Strings;
+import com.google.common.io.CharSink;
+import com.google.common.io.FileWriteMode;
+import com.google.common.io.Files;
 import com.google.inject.Inject;
 import com.radixdlt.application.system.state.EpochData;
 import com.radixdlt.constraintmachine.*;
 import com.radixdlt.crypto.HashUtils;
+import com.radixdlt.properties.RuntimeProperties;
 import com.radixdlt.statecomputer.forks.Forks;
 import com.radixdlt.store.DatabaseEnvironment;
 import com.radixdlt.store.berkeley.BerkeleyAdditionalStore;
 import com.radixdlt.utils.Bytes;
 import com.radixdlt.utils.Longs;
 import com.sleepycat.je.*;
+import java.io.*;
 import java.nio.charset.StandardCharsets;
 import java.util.Optional;
 import java.util.function.Function;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.bouncycastle.util.Arrays;
+import org.bouncycastle.util.encoders.Hex;
 
 public class BerkeleySubstateAccumulatorHashStore implements BerkeleyAdditionalStore {
 
   private static final Logger logger = LogManager.getLogger();
-  protected static final byte[] SUBSTATE_ACCUMULATOR_HASH_KEY =
+  private static final byte[] SUBSTATE_ACCUMULATOR_HASH_KEY =
       "substate_accumulator_hash_key".getBytes(StandardCharsets.UTF_8);
+  private static final String EPOCH_HASH_FILE_PATH =
+      "radixdlt-core/radixdlt/src/main/resources/epoch-hash";
+  public static final String EPOCH_HASH_FILE_SEPARATOR = "=";
 
   private Database substateAccumulatorHashDatabase;
   private Database epochHashDatabase;
 
   private byte[] currentSubstateAccumulatorHash;
```

Comment:
If an exception happens half way through ingesting a batch; this hash could be out of sync with the database, causing incorrect hashes to be written - can we instead just read from the database when we need to instead of having this cached on the class?